toplevel/coqtop.ml

```diff
@@ -50,6 +50,40 @@ let print_memory_stat () =
 
 let _ = at_exit print_memory_stat
 
+let interp_set_option opt v old =
+  let open Goptions in
+  let err expect =
+    let opt = String.concat " " opt in
+    let got = v in (* avoid colliding with Pp.v *)
+    CErrors.user_err
+      Pp.(str "-set: " ++ str opt ++
+          str" expects " ++ str expect ++
+          str" but got " ++ str got)
+  in
+  match old with
+  | BoolValue _ ->
+    let v = match v with
```

Comment:
In the parsing you allow `-set "This Option =true"` but not `-set "This Option = true"`.
IMO you should trim white spaces for all non-string options.

Reply:
(and check if int_of_string does that)

Reply:
done
int_of_string doesn't trim.